extensions/amp-video/0.1/amp-video.js

```diff
@@ -42,6 +42,18 @@ class AmpVideo extends AMP.BaseElement {
       this.video_ = null;
     }
 
+    /**
+     * @param {boolean=} opt_onLayout
+     * @override
+     */
+    preconnectCallback(opt_onLayout) {
+      const videoSrc = this.element.getAttribute('src');
```

Comment:
Do you think it is worth also checking for `<source>` child elements if `src` is not available? basically preconnect to which ever is the first one.

I don't know what the usage of source elements is, but it might be pretty high.